base/MAPL_EsmfRegridder.F90

```diff
@@ -1493,16 +1496,25 @@ subroutine create_route_handle(this, kind, rc)
         end if
         select case (spec%regrid_method)
         case (REGRID_METHOD_BILINEAR, REGRID_METHOD_BILINEAR_MONOTONIC)
-
-           call ESMF_FieldRegridStore(src_field, dst_field, &
-                & regridmethod=ESMF_REGRIDMETHOD_BILINEAR, &
-                & linetype=ESMF_LINETYPE_GREAT_CIRCLE, & ! closer to SJ Lin interpolation weights?
-                & srcTermProcessing = srcTermProcessing, &
-                & factorList=factorList, factorIndexList=factorIndexList, &
-                & routehandle=route_handle, unmappedaction=unmappedaction, rc=status)
-           _VERIFY(status)
+           if (has_mask) then
+              call ESMF_FieldRegridStore(src_field, dst_field, &
+                   & dstMaskValues = [MAPL_MASK_OUT], &
+                   & regridmethod=ESMF_REGRIDMETHOD_BILINEAR, &
+                   & linetype=ESMF_LINETYPE_GREAT_CIRCLE, & ! closer to SJ Lin interpolation weights?
+                   & srcTermProcessing = srcTermProcessing, &
+                   & factorList=factorList, factorIndexList=factorIndexList, &
+                   & routehandle=route_handle, unmappedaction=unmappedaction, _RC)
+           else
+              call ESMF_FieldRegridStore(src_field, dst_field, &
+                   & regridmethod=ESMF_REGRIDMETHOD_BILINEAR, &
+                   & linetype=ESMF_LINETYPE_GREAT_CIRCLE, & ! closer to SJ Lin interpolation weights?
+                   & srcTermProcessing = srcTermProcessing, &
+                   & factorList=factorList, factorIndexList=factorIndexList, &
+                   & routehandle=route_handle, unmappedaction=unmappedaction, _RC)
+           end if
```

Comment:
I will not ask you to make the change, but there is a way to avoid the else/duplication here:
```suggestion
              integer, allocatable :: dstMaskValues(:)
              ...
              if (has_mask) dstMaskValues = [MAPL_MASK_OUT] ! otherwise unallocated
              
              call ESMF_FieldRegridStore(src_field, dst_field, &
                   & dstMaskValues = dstMaskValues, & ! unallocated treated as not PRESENT
                   & regridmethod=ESMF_REGRIDMETHOD_BILINEAR, &
                   & linetype=ESMF_LINETYPE_GREAT_CIRCLE, & ! closer to SJ Lin interpolation weights?
                   & srcTermProcessing = srcTermProcessing, &
                   & factorList=factorList, factorIndexList=factorIndexList, &
                   & routehandle=route_handle, unmappedaction=unmappedaction, _RC)
```

Reply:
@metdyn could clean this up on a subsequent PR. I'll just resolve for now.

Reply:
Thanks, Toms' suggestion works instantly!   I have added this into the next PR.